### PR TITLE
[#9536] Student profile page e2e test migration

### PR DESCRIPTION
**Pull request by jacoblipech**
Part of #9536

- [x] Basic workflow of a student user interacting with student profile page such as uploading, editing profile picture, editing, updating profile details.
- [x] Remove migrated pageobjects such as `NotAuthorizedPage`, `EntityNotFoundPage` originally in v6 as `StudentProfilePicPage` is not available in v7.
- [x] Remove unnecessary data generated in the `StudentProfileE2E` json file

### CODE REVIEW

**Comment by jacoblipech** [comment]
@wkurniawan07 Can I trouble you for an initial review? I am not too sure about the relevancy of the previous `testAjaxPictureUrl` function in v7 and the `getWhetherPictureIsPresentInGcs` method previous. 

The current implementation handles the basic logic of a student logging in and performing interactions with the student profile page

File: src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import teammates.e2e.pageobjects.StudentHomePage;
+import teammates.e2e.pageobjects.StudentProfilePage;
+import teammates.e2e.util.BackDoor;
+import teammates.e2e.util.Priority;
+import teammates.e2e.util.TestProperties;
+
+/**
+ * SUT: {@link teammates.common.util.Const.WebPageURIs#STUDENT_PROFILE_PAGE}.
+ */
+@Priority(-3)
```
Comment (wkurniawan07) [inline src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java]: I don't think you need this.

File: src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        profilePage.showPictureEditor();
+        profilePage.waitForUploadEditModalVisible();
+
+        profilePage.editProfilePhoto();
+        profilePage.ensureProfileContains("short.name", "e@email.tmt", "inst", "American",
+                StudentProfileAttributes.Gender.FEMALE, "this is enough!$%&*</>");
+        profilePage.verifyPhotoSize("295px", "295px");
+
+        JSONObject studentProfileAttributes = new JSONObject(
```
Comment (wkurniawan07) [inline src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java]: Related to my comment in the `BackDoor` class: This shouldn't be a `JSONObject`. It should be the proper `StudentProfileAttributes` object.

File: src/e2e/java/teammates/e2e/pageobjects/StudentProfilePage.java [hunk 3]
```diff
@@ -14,4 +68,161 @@ protected boolean containsExpectedPageContents() {
...
+        click(uploadPopupButton);
+        waitForUploadEditModalVisible();
+    }
+
+    public void verifyPhotoSize(String height, String width) {
+        assertEquals(height, browser.driver.findElement(By.className("cropper")).getCssValue("height"));
+        assertEquals(width, browser.driver.findElement(By.className("cropper")).getCssValue("width"));
+    }
+
+    // this works completely
```
Comment (wkurniawan07) [inline src/e2e/java/teammates/e2e/pageobjects/StudentProfilePage.java]: What does this mean?

**Comment by jacoblipech** [comment]
@wkurniawan07 Sorry for the delay. Made the necessary changes.

Currently, the buttons for flip up/down, flip left/right, rotate left, rotate right are tested. There isn't anyway to verify the edited changes as only the image data changes after each action. I guess the only verification for now is that no error when the click action is made

File: src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * SUT: {@link teammates.common.util.Const.WebPageURIs#STUDENT_PROFILE_PAGE}.
+ */
+public class StudentProfilePageE2ETest extends BaseE2ETestCase {
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/StudentProfilePageE2ETest.json");
+
+        String studentGoogleId = TestProperties.TEST_STUDENT2_ACCOUNT;
```
Comment (wkurniawan07) [inline src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java]: I don't think any of the already-migrated E2E tests need real data, but let's keep it for another day.

Comment (jacoblipech) [reply src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java]: Can I clarify what does don't need real data means? I will try to keep it in mind for future tests

Comment (wkurniawan07) [reply src/e2e/java/teammates/e2e/cases/e2e/StudentProfilePageE2ETest.java]: Real data = actual Google accounts. Even if you examined their counterparts in V6, none of the two migrated actually need real accounts.

**Pull request merged**